From d104160104f50a3b2fd5e311578314e927ea73ca Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 11 Jun 2010 13:05:46 +0200
Subject: [PATCH 013/249] [eel] GSEAL eel-accessibility.

---
 eel/eel-accessibility.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/eel/eel-accessibility.c b/eel/eel-accessibility.c
index 71fd776..5e3fd5a 100644
--- a/eel/eel-accessibility.c
+++ b/eel/eel-accessibility.c
@@ -225,7 +225,7 @@ get_simple_text (gpointer object)
 	EelAccessibleTextIface *aif;
 
 	if (GTK_IS_ACCESSIBLE (object)) {
-		gobject = G_OBJECT (GTK_ACCESSIBLE (object)->widget);
+		gobject = G_OBJECT (gtk_accessible_get_widget (GTK_ACCESSIBLE (object)));
 	} else {
 		gobject = eel_accessibility_get_gobject (object);
 	}
-- 
1.7.2

