From 756beaef00e01c686b061ea77d56c7aa6768d32e Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 11 Jun 2010 13:12:50 +0200
Subject: [PATCH 027/249] [ln-p] GSEAL nautilus-autorun.

---
 libnautilus-private/nautilus-autorun.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libnautilus-private/nautilus-autorun.c b/libnautilus-private/nautilus-autorun.c
index 8d57e31..0760622 100644
--- a/libnautilus-private/nautilus-autorun.c
+++ b/libnautilus-private/nautilus-autorun.c
@@ -911,7 +911,7 @@ show_dialog:
 
 	gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
 	hbox = gtk_hbox_new (FALSE, 12);
-	gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), hbox, TRUE, TRUE, 0);
+	gtk_box_pack_start (GTK_BOX (gtk_dialog_get_content_area (GTK_DIALOG (dialog))), hbox, TRUE, TRUE, 0);
 	gtk_container_set_border_width (GTK_CONTAINER (hbox), 12);
 
 	icon = g_mount_get_icon (mount);
@@ -1030,7 +1030,7 @@ show_dialog:
 		data->should_eject = FALSE;
 	}
 	gtk_dialog_add_action_widget (GTK_DIALOG (dialog), eject_button, AUTORUN_DIALOG_RESPONSE_EJECT);
-	gtk_button_box_set_child_secondary (GTK_BUTTON_BOX (GTK_DIALOG (dialog)->action_area), eject_button, TRUE);
+	gtk_button_box_set_child_secondary (GTK_BUTTON_BOX (gtk_dialog_get_action_area (GTK_DIALOG (dialog))), eject_button, TRUE);
 
 	/* show the dialog */
 	gtk_widget_show_all (dialog);
-- 
1.7.2

