From 3cd134fde66949514eb86b60795394f1aa7a2db6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 11 Jun 2010 13:14:13 +0200
Subject: [PATCH 031/249] [ln-p] GSEAL nautilus-clipboard.

---
 libnautilus-private/nautilus-clipboard.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/libnautilus-private/nautilus-clipboard.c b/libnautilus-private/nautilus-clipboard.c
index 01b1fb1..947dbba 100644
--- a/libnautilus-private/nautilus-clipboard.c
+++ b/libnautilus-private/nautilus-clipboard.c
@@ -585,17 +585,19 @@ nautilus_clipboard_get_uri_list_from_selection_data (GtkSelectionData *selection
 	GList *items;
 	char **lines;
 
-	if (selection_data->type != copied_files_atom
-	    || selection_data->length <= 0) {
+	if (gtk_selection_data_get_data_type (selection_data) != copied_files_atom
+	    || gtk_selection_data_get_length (selection_data) <= 0) {
 		items = NULL;
 	} else {
+		guchar *data;
 		/* Not sure why it's legal to assume there's an extra byte
 		 * past the end of the selection data that it's safe to write
 		 * to. But gtk_editable_selection_received does this, so I
 		 * think it is OK.
 		 */
-		selection_data->data[selection_data->length] = '\0';
-		lines = g_strsplit (selection_data->data, "\n", 0);
+		data = (guchar *) gtk_selection_data_get_data (selection_data);
+		data[gtk_selection_data_get_length (selection_data)] = '\0';
+		lines = g_strsplit (data, "\n", 0);
 		items = convert_lines_to_str_list (lines, cut);
 		g_strfreev (lines);
 	}
-- 
1.7.2

