From dddf609f373f472bca41070296a54be7e67b7adf Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 11 Jun 2010 13:15:07 +0200
Subject: [PATCH 034/249] [ln-p] GSEAL nautilus-file-conflict-dialog.

---
 .../nautilus-file-conflict-dialog.c                |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libnautilus-private/nautilus-file-conflict-dialog.c b/libnautilus-private/nautilus-file-conflict-dialog.c
index ff64933..e6874a5 100644
--- a/libnautilus-private/nautilus-file-conflict-dialog.c
+++ b/libnautilus-private/nautilus-file-conflict-dialog.c
@@ -556,7 +556,7 @@ nautilus_file_conflict_dialog_init (NautilusFileConflictDialog *fcd)
 
 	/* Setup HIG properties */
 	gtk_container_set_border_width (GTK_CONTAINER (dialog), 5);
-	gtk_box_set_spacing (GTK_BOX (dialog->vbox), 14);
+	gtk_box_set_spacing (GTK_BOX (gtk_dialog_get_content_area (dialog)), 14);
 	gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
 	gtk_dialog_set_has_separator (dialog, FALSE);
 
-- 
1.7.2

