From 2206f6c82c5a49769ccdede84d74bf595a0a57c1 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 11 Jun 2010 13:15:20 +0200
Subject: [PATCH 035/249] [ln-p] GSEAL nautilus-file-operations.

---
 libnautilus-private/nautilus-file-operations.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libnautilus-private/nautilus-file-operations.c b/libnautilus-private/nautilus-file-operations.c
index 52bc77f..1e13778 100644
--- a/libnautilus-private/nautilus-file-operations.c
+++ b/libnautilus-private/nautilus-file-operations.c
@@ -2204,7 +2204,7 @@ prompt_empty_trash (GtkWindow *parent_window)
 	/* Make transient for the window group */
 	gtk_widget_realize (dialog);
 	if (screen != NULL) {
-		gdk_window_set_transient_for (GTK_WIDGET (dialog)->window,
+		gdk_window_set_transient_for (gtk_widget_get_window (GTK_WIDGET (dialog)),
 				      		gdk_screen_get_root_window (screen));
 	}
 	
-- 
1.7.2

