From a0d1ca75e37fd26135fe9c85d8d86dc75e831d86 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:21:45 +0200
Subject: [PATCH 059/249] [src] GSEAL nautilus-location-dialog.

---
 src/nautilus-location-dialog.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nautilus-location-dialog.c b/src/nautilus-location-dialog.c
index ce38466..5cd20fa 100644
--- a/src/nautilus-location-dialog.c
+++ b/src/nautilus-location-dialog.c
@@ -168,7 +168,7 @@ nautilus_location_dialog_init (NautilusLocationDialog *dialog)
 	gtk_window_set_destroy_with_parent (GTK_WINDOW (dialog), TRUE);
 	gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
 	gtk_container_set_border_width (GTK_CONTAINER (dialog), 5);
-	gtk_box_set_spacing (GTK_BOX (GTK_DIALOG (dialog)->vbox), 2);
+	gtk_box_set_spacing (GTK_BOX (gtk_dialog_get_content_area (GTK_DIALOG (dialog))), 2);
 	
 	box = gtk_hbox_new (FALSE, 12);
 	gtk_container_set_border_width (GTK_CONTAINER (box), 5);
@@ -192,7 +192,7 @@ nautilus_location_dialog_init (NautilusLocationDialog *dialog)
 	gtk_box_pack_start (GTK_BOX (box), dialog->details->entry, 
 			    TRUE, TRUE, 0);
 	
-	gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox),
+	gtk_box_pack_start (GTK_BOX (gtk_dialog_get_content_area (GTK_DIALOG (dialog))),
 			    box, FALSE, TRUE, 0);
 
 	gtk_dialog_add_button (GTK_DIALOG (dialog),
-- 
1.7.2

