From cb1e2694bdaa8c8fbad8b2541f0efe354d81b1c0 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:21:53 +0200
Subject: [PATCH 060/249] [src] GSEAL nautilus-navigation-window-menus.

---
 src/nautilus-navigation-window-menus.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/nautilus-navigation-window-menus.c b/src/nautilus-navigation-window-menus.c
index 6ab44a3..cee948f 100644
--- a/src/nautilus-navigation-window-menus.c
+++ b/src/nautilus-navigation-window-menus.c
@@ -398,7 +398,7 @@ connect_proxy_cb (GtkActionGroup *action_group,
 	if (!GTK_IS_MENU_ITEM (proxy))
 		return;
 
-	label = GTK_LABEL (GTK_BIN (proxy)->child);
+	label = GTK_LABEL (gtk_bin_get_child (GTK_BIN (proxy)));
 
 	gtk_label_set_use_underline (label, FALSE);
 	gtk_label_set_ellipsize (label, PANGO_ELLIPSIZE_END);
-- 
1.7.2

