From b5407396039035e6f4e78c2d9d03891d76dd93bd Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:23:38 +0200
Subject: [PATCH 070/249] [src] GSEAL nautilus-window-bookmarks.

---
 src/nautilus-window-bookmarks.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/nautilus-window-bookmarks.c b/src/nautilus-window-bookmarks.c
index 0d25e35..84d0fa0 100644
--- a/src/nautilus-window-bookmarks.c
+++ b/src/nautilus-window-bookmarks.c
@@ -195,7 +195,7 @@ connect_proxy_cb (GtkActionGroup *action_group,
 	if (!GTK_IS_MENU_ITEM (proxy))
 		return;
 
-	label = GTK_LABEL (GTK_BIN (proxy)->child);
+	label = GTK_LABEL (gtk_bin_get_child (GTK_BIN (proxy)));
 
 	gtk_label_set_use_underline (label, FALSE);
 	gtk_label_set_ellipsize (label, PANGO_ELLIPSIZE_END);
-- 
1.7.2

