From 44af78c3c205f88801a8bdfa3cb0a02f7cd87d3f Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:23:47 +0200
Subject: [PATCH 071/249] [src] GSEAL nautilus-window-manage-views.

---
 src/nautilus-window-manage-views.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/nautilus-window-manage-views.c b/src/nautilus-window-manage-views.c
index f982c39..3ec3157 100644
--- a/src/nautilus-window-manage-views.c
+++ b/src/nautilus-window-manage-views.c
@@ -1399,7 +1399,7 @@ location_has_really_changed (NautilusWindowSlot *slot)
 	if (slot->new_content_view != NULL) {
 		widget = nautilus_view_get_widget (slot->new_content_view);
 		/* Switch to the new content view. */
-		if (widget->parent == NULL) {
+		if (gtk_widget_get_parent (widget) == NULL) {
 			if (slot->content_view != NULL) {
 				nautilus_window_slot_disconnect_content_view (slot, slot->content_view);
 			}
-- 
1.7.2

