From bbeb350ef219f6e94d9eba6e78d0ebbdb2201a08 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:23:58 +0200
Subject: [PATCH 072/249] [src] GSEAL nautilus-window.

---
 src/nautilus-window.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nautilus-window.c b/src/nautilus-window.c
index 939ee11..588464e 100644
--- a/src/nautilus-window.c
+++ b/src/nautilus-window.c
@@ -333,10 +333,10 @@ update_cursor (NautilusWindow *window)
 
 	if (slot->allow_stop) {
 		cursor = gdk_cursor_new (GDK_WATCH);
-		gdk_window_set_cursor (GTK_WIDGET (window)->window, cursor);
+		gdk_window_set_cursor (gtk_widget_get_window (GTK_WIDGET (window)), cursor);
 		gdk_cursor_unref (cursor);
 	} else {
-		gdk_window_set_cursor (GTK_WIDGET (window)->window, NULL);
+		gdk_window_set_cursor (gtk_widget_get_window (GTK_WIDGET (window)), NULL);
 	}
 }
 
-- 
1.7.2

