From d9f80b3f407b65f559c0ca47cddac41175ac353d Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 Jun 2010 13:24:20 +0200
Subject: [PATCH 074/249] [test] GSEAL test-eel-image-table.

---
 test/test-eel-image-table.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/test/test-eel-image-table.c b/test/test-eel-image-table.c
index 267ac15..271642b 100644
--- a/test/test-eel-image-table.c
+++ b/test/test-eel-image-table.c
@@ -179,6 +179,7 @@ image_table_size_allocate (GtkWidget *image_table,
 			   gpointer callback_data)
 {
 	static int recursion_count = 0;
+	GtkAllocation w_allocation;
 
 	g_return_if_fail (EEL_IS_IMAGE_TABLE (image_table));
 	g_return_if_fail (allocation != NULL);
@@ -190,8 +191,9 @@ image_table_size_allocate (GtkWidget *image_table,
 
 	/*gtk_widget_queue_resize (GTK_WIDGET (callback_data));*/
 
+	gtk_widget_get_allocation (GTK_WIDGET (image_table), &w_allocation);
 	if (0) gtk_widget_size_allocate (GTK_WIDGET (image_table),
-					 &GTK_WIDGET (image_table)->allocation);
+					 &w_allocation);
 	
 	g_print ("%s(%d)\n", G_STRFUNC, recursion_count);
 
-- 
1.7.2

