diff -up nautilus-2.31.3/libnautilus-private/apps_nautilus_preferences.schemas.in.bool nautilus-2.31.3/libnautilus-private/apps_nautilus_preferences.schemas.in
--- nautilus-2.31.3/libnautilus-private/apps_nautilus_preferences.schemas.in.bool	2010-06-18 23:58:49.144816828 -0400
+++ nautilus-2.31.3/libnautilus-private/apps_nautilus_preferences.schemas.in	2010-06-18 23:59:04.839067421 -0400
@@ -220,7 +220,7 @@
       <key>/schemas/apps/nautilus/preferences/navigation_window_saved_maximized</key>
       <applyto>/apps/nautilus/preferences/navigation_window_saved_maximized</applyto>
       <owner>nautilus</owner>
-      <type>boolean</type>
+      <type>bool</type>
       <default></default>
       <locale name="C">
          <short>Whether the navigation window should be maximized.</short>
