diff -up nautilus-2.21.1/libnautilus-extension/libnautilus-extension.pc.in.extensiondir nautilus-2.21.1/libnautilus-extension/libnautilus-extension.pc.in
--- nautilus-2.21.1/libnautilus-extension/libnautilus-extension.pc.in.extensiondir	2007-12-23 19:19:33.000000000 -0500
+++ nautilus-2.21.1/libnautilus-extension/libnautilus-extension.pc.in	2007-12-23 19:20:04.000000000 -0500
@@ -2,7 +2,7 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
-extensiondir=@libdir@/nautilus/extensions-1.0
+extensiondir=@libdir@/nautilus/extensions-2.0
 
 Name: libnautilus-extension
 Description: A library to create Nautilus view extensions
