diff -up nautilus-2.23.5.1/src/nautilus-file-management-properties.c.icon-names nautilus-2.23.5.1/src/nautilus-file-management-properties.c
--- nautilus-2.23.5.1/src/nautilus-file-management-properties.c.icon-names	2008-07-21 05:43:25.000000000 -0400
+++ nautilus-2.23.5.1/src/nautilus-file-management-properties.c	2008-07-27 22:58:28.000000000 -0400
@@ -801,7 +801,7 @@ nautilus_file_management_properties_dial
 			       (GClosureNotify)g_object_unref,
 			       0);
 
-	gtk_window_set_icon_name (GTK_WINDOW (dialog), "file-manager");
+	gtk_window_set_icon_name (GTK_WINDOW (dialog), "system-file-manager");
 
 	if (window) {
 		gtk_window_set_screen (GTK_WINDOW (dialog), gtk_window_get_screen(window));
diff -up nautilus-2.23.5.1/src/nautilus-navigation-window.c.icon-names nautilus-2.23.5.1/src/nautilus-navigation-window.c
--- nautilus-2.23.5.1/src/nautilus-navigation-window.c.icon-names	2008-07-27 22:58:53.000000000 -0400
+++ nautilus-2.23.5.1/src/nautilus-navigation-window.c	2008-07-27 22:59:23.000000000 -0400
@@ -1221,7 +1221,7 @@ static NautilusIconInfo *
 real_get_icon (NautilusWindow *window,
 	       NautilusWindowSlot *slot)
 {
-	return nautilus_icon_info_lookup_from_name ("file-manager", 48);
+	return nautilus_icon_info_lookup_from_name ("system-file-manager", 48);
 }
 
 static void
