diff -up nautilus-2.20.0/src/nautilus-sidebar-title.c.small-font nautilus-2.20.0/src/nautilus-sidebar-title.c
--- nautilus-2.20.0/src/nautilus-sidebar-title.c.small-font	2007-10-02 21:04:51.000000000 -0400
+++ nautilus-2.20.0/src/nautilus-sidebar-title.c	2007-10-02 21:16:36.000000000 -0400
@@ -364,7 +364,7 @@ update_title_font (NautilusSidebarTitle 
 	title_font = pango_font_description_copy (style->font_desc);
 
 	max_style_font_size = pango_font_description_get_size (title_font) * 1.8 / PANGO_SCALE;
-	if (max_style_font_size < MIN_TITLE_FONT_SIZE) {
+	if (max_style_font_size < MIN_TITLE_FONT_SIZE + 1) {
 		max_style_font_size = MIN_TITLE_FONT_SIZE + 1;
 	}
 
