commit c926b072165b822af8dfd6f5d8a413809bb6b541
Author: Cosimo Cecchi <cosimoc@gnome.org>
Date:   Sat Feb 5 14:40:15 2011 -0500

    bookmark: apply name and custom-name during construction

diff --git a/libnautilus-private/nautilus-bookmark.c b/libnautilus-private/nautilus-bookmark.c
index 7bc26f5..426f935 100644
--- a/libnautilus-private/nautilus-bookmark.c
+++ b/libnautilus-private/nautilus-bookmark.c
@@ -387,14 +387,14 @@ nautilus_bookmark_class_init (NautilusBookmarkClass *class)
 				     "Bookmark's name",
 				     "The name of this bookmark",
 				     NULL,
-				     G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+				     G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT);
 
 	properties[PROP_CUSTOM_NAME] =
 		g_param_spec_boolean ("custom-name",
 				      "Whether the bookmark has a custom name",
 				      "Whether the bookmark has a custom name",
 				      FALSE,
-				      G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+				      G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT);
 
 	properties[PROP_LOCATION] =
 		g_param_spec_object ("location",
