diff -up nautilus-2.27.91/src/nautilus-file-management-properties.ui.dupe-id nautilus-2.27.91/src/nautilus-file-management-properties.ui
--- nautilus-2.27.91/src/nautilus-file-management-properties.ui.dupe-id	2009-08-26 20:04:20.492274436 -0400
+++ nautilus-2.27.91/src/nautilus-file-management-properties.ui	2009-08-26 20:04:26.611029555 -0400
@@ -1128,7 +1128,7 @@
                               </packing>
                             </child>
                             <child>
-                              <object class="GtkLabel" id="label">
+                              <object class="GtkLabel" id="label99">
                                 <property name="height_request">6</property>
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes"/>
