Index: libnautilus-private/nautilus-dnd.c
===================================================================
--- libnautilus-private/nautilus-dnd.c	(révision 14260)
+++ libnautilus-private/nautilus-dnd.c	(copie de travail)
@@ -422,7 +422,7 @@
 				*action = GDK_ACTION_MOVE;
 			}
 			
-			g_free (target);
+			g_object_unref (target);
 			nautilus_file_unref (dropped_file);
 			nautilus_file_unref (target_file);
 			return;
