--- nautilus-2.14.1/src/nautilus-window-menus.c.about	2006-05-12 00:16:23.000000000 -0400
+++ nautilus-2.14.1/src/nautilus-window-menus.c	2006-05-12 00:17:54.000000000 -0400
@@ -482,6 +482,9 @@
 
 		gtk_window_set_transient_for (GTK_WINDOW (about), GTK_WINDOW (user_data));
 
+		g_signal_connect (G_OBJECT (about), "response", 
+				  G_CALLBACK (gtk_widget_destroy), NULL);
+
 		eel_add_weak_pointer (&about);
 	}
 	
