Index: libnautilus-private/nautilus-mime-actions.c
===================================================================
--- libnautilus-private/nautilus-mime-actions.c	(revision 14785)
+++ libnautilus-private/nautilus-mime-actions.c	(working copy)
@@ -1327,6 +1327,7 @@
 	NautilusFile *file;
 	GFile *location;
 	GMountOperation *mount_op;
+	GList *l, *next;
 
 	if (parameters->not_mounted != NULL) {
 		file = parameters->not_mounted->data;
@@ -1348,6 +1349,14 @@
 		return;
 	}
 	
+	/*  once the mount is finished, refresh all attributes        */
+	/*  - fixes new windows not appearing after successful mount  */
+	for (l = parameters->files; l != NULL; l = next) {
+		file = NAUTILUS_FILE (l->data);
+		next = l->next;
+		nautilus_file_invalidate_all_attributes (file);
+	}
+	
 	nautilus_file_list_call_when_ready
 		(parameters->files,
 		 nautilus_mime_actions_get_required_file_attributes () | NAUTILUS_FILE_ATTRIBUTE_LINK_INFO,
