--- trunk/libnautilus-private/apps_nautilus_preferences.schemas.in	2008/08/18 20:47:18	14489
+++ trunk/libnautilus-private/apps_nautilus_preferences.schemas.in	2008/08/20 22:51:57	14508
@@ -1092,7 +1092,7 @@
     <schema>
       <key>/schemas/apps/nautilus/desktop/text_ellipsis_limit</key>
       <applyto>/apps/nautilus/desktop/text_ellipsis_limit</applyto>
-      <type>integer</type>
+      <type>int</type>
       <default>3</default>
       <locale name="C">
          <short>Text Ellipsis Limit</short>
