diff -up nautilus-2.27.4/src/nautilus-spatial-window.c.menu-icons3 nautilus-2.27.4/src/nautilus-spatial-window.c
--- nautilus-2.27.4/src/nautilus-spatial-window.c.menu-icons3	2009-08-03 18:59:28.015377172 -0400
+++ nautilus-2.27.4/src/nautilus-spatial-window.c	2009-08-03 19:08:19.623627208 -0400
@@ -683,6 +683,7 @@ location_button_clicked_callback (GtkWid
 
 		name = nautilus_file_get_display_name (file);
 		menu_item = gtk_image_menu_item_new_with_label (name);
+		gtk_image_menu_item_set_always_show_image (GTK_IMAGE_MENU_ITEM (menu_item), TRUE);
 		g_free (name);
 
 		if (first_item == NULL) {
@@ -941,6 +942,14 @@ static const GtkActionEntry spatial_entr
                                  G_CALLBACK (action_search_callback) },
 };
 
+static const char* icon_entries[] = {
+	"/MenuBar/Other Menus/Places/Home",
+	"/MenuBar/Other Menus/Places/Go to Computer",
+	"/MenuBar/Other Menus/Places/Go to Templates",
+	"/MenuBar/Other Menus/Places/Go to Trash",
+	"/MenuBar/Other Menus/Places/Go to Network"
+};
+
 static void
 nautilus_spatial_window_init (NautilusSpatialWindow *window)
 {
@@ -951,6 +960,8 @@ nautilus_spatial_window_init (NautilusSp
 	GtkUIManager *ui_manager;
 	GtkTargetList *targets;
 	const char *ui;
+	int i;
+	GtkWidget *menuitem;
 
 	window->details = G_TYPE_INSTANCE_GET_PRIVATE (window,
 						       NAUTILUS_TYPE_SPATIAL_WINDOW,
@@ -1042,6 +1053,11 @@ nautilus_spatial_window_init (NautilusSp
 	
 	ui = nautilus_ui_string_get ("nautilus-spatial-window-ui.xml");
 	gtk_ui_manager_add_ui_from_string (ui_manager, ui, -1, NULL);
+
+	for (i = 0; i < G_N_ELEMENTS (icon_entries); i++) {
+		menuitem = gtk_ui_manager_get_widget (ui_manager, icon_entries[i]);
+		gtk_image_menu_item_set_always_show_image (GTK_IMAGE_MENU_ITEM (menuitem), TRUE);
+	}
 }
 
 static void
