--- nautilus-2.16.2/libnautilus-private/nautilus-search-engine.c.order	2006-11-22 13:19:49.000000000 +0100
+++ nautilus-2.16.2/libnautilus-private/nautilus-search-engine.c	2006-11-22 13:20:03.000000000 +0100
@@ -126,19 +126,20 @@
 {
 	NautilusSearchEngine *engine;
 	
-#ifdef HAVE_TRACKER
-	engine = nautilus_search_engine_tracker_new ();
+#ifdef HAVE_BEAGLE
+	engine = nautilus_search_engine_beagle_new ();
 	if (engine) {
 		return engine;
 	}
 #endif
-
-#ifdef HAVE_BEAGLE
-	engine = nautilus_search_engine_beagle_new ();
+	
+#ifdef HAVE_TRACKER
+	engine = nautilus_search_engine_tracker_new ();
 	if (engine) {
 		return engine;
 	}
 #endif
+
 	engine = nautilus_search_engine_simple_new ();
 	return engine;
 }
