--- nautilus-2.5.7/libnautilus-private/apps_nautilus_preferences.schemas.in.rhconfig	2004-02-13 09:14:16.000000000 +0100
+++ nautilus-2.5.7/libnautilus-private/apps_nautilus_preferences.schemas.in	2004-02-13 09:17:23.000000000 +0100
@@ -144,7 +144,7 @@
       <applyto>/apps/nautilus/preferences/theme</applyto>
       <owner>nautilus</owner>
       <type>string</type>
-      <default>default</default>
+      <default>Bluecurve</default>
       <locale name="C">
          <short>Current Nautilus theme (deprecated)</short>
          <long>
