--- nautilus-autorun-software.desktop.in~	2008-04-02 06:53:05.000000000 -0400
+++ nautilus-autorun-software.desktop.in	2008-04-17 20:08:52.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=Autorun Prompt
 TryExec=nautilus-autorun-software
 Exec=nautilus-autorun-software %u
+X-Gnome-Vfs-System=gio
 Icon=application-x-executable
 NoDisplay=true
 Terminal=false
--- src/network-scheme.desktop.in~	2008-03-29 05:40:25.000000000 -0400
+++ src/network-scheme.desktop.in	2008-04-17 20:08:48.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Browse bookmarked and local network locations
 TryExec=nautilus
 Exec=nautilus --no-desktop network:
+X-Gnome-Vfs-System=gio
 Terminal=false
 StartupNotify=true
 Type=Application
--- nautilus.desktop.in.in~	2008-03-29 05:40:44.000000000 -0400
+++ nautilus.desktop.in.in	2008-04-17 20:08:45.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Browse the file system with the file manager
 TryExec=nautilus
 Exec=nautilus --no-desktop --browser %U
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
--- nautilus-folder-handler.desktop.in.in~	2008-03-29 05:40:44.000000000 -0400
+++ nautilus-folder-handler.desktop.in.in	2008-04-17 20:08:41.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=Open Folder
 TryExec=nautilus
 Exec=nautilus --no-desktop %U
+X-Gnome-Vfs-System=gio
 NoDisplay=true
 Terminal=false
 Icon=folder-open
--- nautilus.desktop~	2008-04-07 15:57:49.000000000 -0400
+++ nautilus.desktop	2008-04-17 20:08:35.000000000 -0400
@@ -150,6 +150,7 @@
 Comment[zh_TW]=使用檔案總管瀏覽檔案系統
 TryExec=nautilus
 Exec=nautilus --no-desktop --browser %U
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
--- nautilus-autorun-software.desktop.in.in~	2008-03-29 05:40:44.000000000 -0400
+++ nautilus-autorun-software.desktop.in.in	2008-04-17 20:08:29.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=Autorun Prompt
 TryExec=nautilus-autorun-software
 Exec=nautilus-autorun-software %u
+X-Gnome-Vfs-System=gio
 Icon=application-x-executable
 NoDisplay=true
 Terminal=false
--- nautilus.desktop.in~	2008-04-02 06:53:04.000000000 -0400
+++ nautilus.desktop.in	2008-04-17 20:08:23.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Browse the file system with the file manager
 TryExec=nautilus
 Exec=nautilus --no-desktop --browser %U
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
--- nautilus-home.desktop.in~	2008-04-02 06:53:04.000000000 -0400
+++ nautilus-home.desktop.in	2008-04-17 20:08:20.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Open your personal folder
 TryExec=nautilus
 Exec=nautilus --no-desktop
+X-Gnome-Vfs-System=gio
 Icon=user-home
 Terminal=false
 StartupNotify=true
--- nautilus-computer.desktop.in.in~	2008-03-29 05:40:44.000000000 -0400
+++ nautilus-computer.desktop.in.in	2008-04-17 20:08:17.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Browse all local and remote disks and folders accessible from this computer
 TryExec=nautilus
 Exec=nautilus --no-desktop computer:
+X-Gnome-Vfs-System=gio
 Icon=computer
 Terminal=false
 StartupNotify=true
--- nautilus-home.desktop.in.in~	2008-03-29 05:40:44.000000000 -0400
+++ nautilus-home.desktop.in.in	2008-04-17 20:08:14.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Open your personal folder
 TryExec=nautilus
 Exec=nautilus --no-desktop
+X-Gnome-Vfs-System=gio
 Icon=user-home
 Terminal=false
 StartupNotify=true
--- nautilus-computer.desktop.in~	2008-04-02 06:53:04.000000000 -0400
+++ nautilus-computer.desktop.in	2008-04-17 20:08:11.000000000 -0400
@@ -4,6 +4,7 @@
 _Comment=Browse all local and remote disks and folders accessible from this computer
 TryExec=nautilus
 Exec=nautilus --no-desktop computer:
+X-Gnome-Vfs-System=gio
 Icon=computer
 Terminal=false
 StartupNotify=true
--- nautilus-file-management-properties.desktop.in~	2008-04-02 06:53:04.000000000 -0400
+++ nautilus-file-management-properties.desktop.in	2008-04-17 20:08:07.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=File Management
 _Comment=Change the behaviour and appearance of file manager windows
 Exec=nautilus-file-management-properties
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
--- nautilus-folder-handler.desktop.in~	2008-04-02 06:53:04.000000000 -0400
+++ nautilus-folder-handler.desktop.in	2008-04-17 20:08:04.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=Open Folder
 TryExec=nautilus
 Exec=nautilus --no-desktop %U
+X-Gnome-Vfs-System=gio
 NoDisplay=true
 Terminal=false
 Icon=folder-open
--- nautilus-file-management-properties.desktop.in.in~	2008-03-29 05:40:45.000000000 -0400
+++ nautilus-file-management-properties.desktop.in.in	2008-04-17 20:08:00.000000000 -0400
@@ -3,6 +3,7 @@
 _Name=File Management
 _Comment=Change the behaviour and appearance of file manager windows
 Exec=nautilus-file-management-properties
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
