diff -up nautilus-2.23.1/libnautilus-private/nautilus-icon-container.c.rtl-fix nautilus-2.23.1/libnautilus-private/nautilus-icon-container.c
--- nautilus-2.23.1/libnautilus-private/nautilus-icon-container.c.rtl-fix	2008-04-21 22:54:48.000000000 +0200
+++ nautilus-2.23.1/libnautilus-private/nautilus-icon-container.c	2008-04-23 17:17:37.356353323 +0200
@@ -8445,6 +8445,25 @@ nautilus_icon_container_accessible_get_t
         return type;
 }
 
+gboolean
+nautilus_icon_container_is_layout_rtl (NautilusIconContainer *container)
+{
+	g_return_val_if_fail (NAUTILUS_IS_ICON_CONTAINER (container), 0);
+
+	return container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_R_L ||
+		container->details->layout_mode == NAUTILUS_ICON_LAYOUT_R_L_T_B;
+}
+
+gboolean
+nautilus_icon_container_is_layout_vertical (NautilusIconContainer *container)
+{
+	g_return_val_if_fail (NAUTILUS_IS_ICON_CONTAINER (container), FALSE);
+
+	return (container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_L_R ||
+		container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_R_L);
+}
+
+
 #if ! defined (NAUTILUS_OMIT_SELF_CHECK)
 
 static char *
@@ -8479,23 +8498,4 @@ nautilus_self_check_icon_container (void
 	EEL_CHECK_STRING_RESULT (check_compute_stretch (100, 100, 64, 105, 105, 40, 40), "35,35:129");
 }
 
-gboolean
-nautilus_icon_container_is_layout_rtl (NautilusIconContainer *container)
-{
-	g_return_val_if_fail (NAUTILUS_IS_ICON_CONTAINER (container), 0);
-
-	return container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_R_L ||
-		container->details->layout_mode == NAUTILUS_ICON_LAYOUT_R_L_T_B;
-}
-
-gboolean
-nautilus_icon_container_is_layout_vertical (NautilusIconContainer *container)
-{
-	g_return_val_if_fail (NAUTILUS_IS_ICON_CONTAINER (container), FALSE);
-
-	return (container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_L_R ||
-		container->details->layout_mode == NAUTILUS_ICON_LAYOUT_T_B_R_L);
-}
-
-
 #endif /* ! NAUTILUS_OMIT_SELF_CHECK */
