Index: libnautilus-private/nautilus-directory-background.c
===================================================================
--- libnautilus-private/nautilus-directory-background.c	(revision 14544)
+++ libnautilus-private/nautilus-directory-background.c	(working copy)
@@ -115,6 +115,7 @@
         if (tmp != NULL) {
                 if (!gdk_color_parse (tmp, color))
                         gdk_color_parse ("black", color);
+		g_free (tmp);
         }
         else {
                 gdk_color_parse ("black", color);
@@ -198,6 +199,7 @@
         else {
                 *placement = EEL_BACKGROUND_CENTERED;
         }
+        g_free (tmp);
                 
         /* Get the color */
         tmp = gconf_client_get_string (client, BG_PREFERENCES_COLOR_SHADING_TYPE, NULL);
@@ -223,6 +225,7 @@
                 use_gradient = FALSE;
                 is_horizontal = FALSE;
         }
+        g_free (tmp);
         
         read_color (client, BG_PREFERENCES_PRIMARY_COLOR, &primary);
         read_color (client, BG_PREFERENCES_SECONDARY_COLOR, &secondary);
