diff -up nautilus-2.27.91/libnautilus-private/nautilus-file-utilities.c.release-your-inhibitions nautilus-2.27.91/libnautilus-private/nautilus-file-utilities.c
--- nautilus-2.27.91/libnautilus-private/nautilus-file-utilities.c.release-your-inhibitions	2009-09-06 21:25:01.462765511 -0400
+++ nautilus-2.27.91/libnautilus-private/nautilus-file-utilities.c	2009-09-06 21:25:34.998778724 -0400
@@ -1100,7 +1100,7 @@ nautilus_uninhibit_power_manager (gint c
 {
 	DBusGProxy *proxy;
 	GError *error;
-	g_return_if_fail (cookie < 0);
+	g_return_if_fail (cookie > 0);
 
 	proxy = get_power_manager_proxy ();
 
